e2e/testcases/otel_collector_test.go

```diff
@@ -179,32 +214,84 @@ func TestOtelCollectorDeployment(t *testing.T) {
 
 	// retry for 2 minutes until metric is accessible from GCM
 	_, err = retry.Retry(120*time.Second, func() error {
-		for _, metricType := range GCMMetricTypes {
+		var err error
+		for _, metricType := range DefaultGCMMetricTypes {
 			descriptor := fmt.Sprintf("%s/%s", GCMMetricPrefix, metricType)
 			it := listMetricInGCM(ctx, nt, client, startTime, descriptor)
-			return validateMetricInGCM(nt, it, descriptor, nt.ClusterName)
+			err = multierr.Append(err, validateMetricInGCM(nt, it, descriptor, nt.ClusterName))
 		}
-		return nil
+		return err
 	})
 	if err != nil {
 		nt.T.Fatal(err)
 	}
 
 	nt.T.Log("Checking the otel-collector log contains no failure...")
-	err = validateDeploymentLogHasNoFailure(nt, ocmetrics.OtelCollectorName, configmanagement.MonitoringNamespace, MetricExportErrorCaption)
+	err = validateDeploymentLogHasNoFailure(nt, csmetrics.OtelCollectorName, configmanagement.MonitoringNamespace, MetricExportErrorCaption)
 	if err != nil {
 		nt.T.Fatal(err)
 	}
 
 	nt.T.Log("Apply custom otel-collector ConfigMap that could cause Monarch label rejected error")
 	nt.MustKubectl("apply", "-f", "../testdata/otel-collector/otel-cm-kustomize-rejected-labels.yaml")
-	if err := nt.Watcher.WatchForCurrentStatus(kinds.Deployment(), ocmetrics.OtelCollectorName, configmanagement.MonitoringNamespace); err != nil {
+	if err := nt.Watcher.WatchForCurrentStatus(kinds.Deployment(), csmetrics.OtelCollectorName, configmanagement.MonitoringNamespace); err != nil {
 		nt.T.Fatal(err)
 	}
 
 	nt.T.Log("Checking the otel-collector log contains failure...")
 	_, err = retry.Retry(60*time.Second, func() error {
-		return validateDeploymentLogHasFailure(nt, ocmetrics.OtelCollectorName, configmanagement.MonitoringNamespace, UnrecognizedLabelErrorCaption)
+		return validateDeploymentLogHasFailure(nt, csmetrics.OtelCollectorName, configmanagement.MonitoringNamespace, UnrecognizedLabelErrorCaption)
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+}
+
+func TestGCMMetrics(t *testing.T) {
+	nt := nomostest.New(t,
+		nomostesting.Reconciliation1,
+		ntopts.RequireGKE(t),
+		ntopts.Unstructured,
+	)
+	nt.T.Cleanup(func() {
+		if t.Failed() {
+			nt.PodLogs("config-management-monitoring", csmetrics.OtelCollectorName, "", false)
+		}
+	})
+	setupMetricsServiceAccount(nt)
+	nt.T.Cleanup(func() {
+		nt.MustKubectl("delete", "cm", csmetrics.OtelCollectorCustomCM, "-n", configmanagement.MonitoringNamespace, "--ignore-not-found")
+		nt.T.Log("Restart otel-collector pod to reset the ConfigMap and log")
+		nomostest.DeletePodByLabel(nt, "app", csmetrics.OpenTelemetry, false)
```

Comment:
Thanks for pointing it out, seems like other tests are also falsely. The creation / deletion of the test ConfigMap otel-collector-custom would trigger the otel-collector pod to restart so that's why they were passing, even with the DeletePodByLabel is no-op. Cleaned up for all.

Reply:
>  The creation / deletion of the test ConfigMap otel-collector-custom would trigger the otel-collector pod to restart so that's why they were passing, even with the DeletePodByLabel is no-op.

Oh, that's right. The otel_controller watches for the ConfigMap and reconciles the otel-collector Deployment.